File: 11_Container_With_Most_Water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        result, l, r = 0, 0, len(height) - 1

        while l < r:
            if height[l] <= height[r]:
                area = height[l] * (r - l)
                l += 1
            else:
                area = height[r] * (r - l)
                r -= 1
            if result < area: result = area

        return result

File: 11_Container_With_Most_Water/solution.c
int maxArea(int* height, int heightSize) {
    int result = 0, l = 0, r = heightSize - 1, area;

    while (l < r) {
        area = (height[l] < height[r] ? height[l] : height[r]) * (r - l);
        if (area > result) result = area;
        if (height[l] < height[r]) l++; else r--;
    }
    return result;
}
